src/rust.rs

```diff
@@ -0,0 +1,201 @@
+    let idf_tool_xtensa_elf_clang = format!("{}/{}-{}", get_tool_path("xtensa-esp32-elf-clang".to_string()), llvm_release, arch);
+
+    RustToolchain {
+        arch: arch.to_string(),
+        llvm_release,
+        llvm_arch,
+        artifact_file_extension,
+        version: "1.55.0-dev".to_string(),
+        rust_dist,
+        rust_src_dist: format!("rust-src-{}", version),
+        rust_dist_file,
+        rust_dist_url,
+        destination_dir: format!("{}/.rustup/toolchains/esp", home_dir().unwrap().display().to_string()),
+        llvm_file,
+        llvm_url,
+        idf_tool_xtensa_elf_clang
+    }
+}
+
+#[cfg(windows)]
+fn set_env_variable(key:&str, value:String) {
+    use winreg::{enums::HKEY_CURRENT_USER, RegKey};
+    let hkcu = RegKey::predef(HKEY_CURRENT_USER);
+    let (env, _) = hkcu.create_subkey("Environment").unwrap(); // create_subkey opens with write permissions
+    env.set_value(key, &value).unwrap();
+}
+
+#[cfg(windows)]
+fn update_env_path(value: &str) {
+    let path_key = "PATH";
+    use winreg::{enums::HKEY_CURRENT_USER, RegKey};
+    let hkcu = RegKey::predef(HKEY_CURRENT_USER);
+    let env = hkcu.open_subkey("Environment").unwrap();
+    let env_path:String = env.get_value(path_key).unwrap();
+    if !env_path.contains(&value) {
+        let updated_env_path = format!("{};{}", env_path, value);
+        set_env_variable(path_key, updated_env_path);
+    }
+}
+
+#[cfg(unix)]
+fn set_env_variable(key:&str, value:&str) {
+
+}
+
+
+fn install_rust_toolchain(toolchain:&RustToolchain) {
+    if Path::new(toolchain.destination_dir.as_str()).exists() {
+        println!("Previous installation of Rust Toolchain exist in: {}", toolchain.destination_dir);
+        println!("Please, remove the directory before new installation.");
+    } else {
+        // create_dir_all(toolchain_destination_dir.clone());
+        prepare_package_strip_prefix(&toolchain.rust_dist_url,
+                                     &toolchain.rust_dist_file,
+                                     toolchain.destination_dir.to_string(),
+                                     "esp");
+    }
+
+    if Path::new(toolchain.idf_tool_xtensa_elf_clang.as_str()).exists() {
+        println!("Previous installation of LLVM exist in: {}", toolchain.idf_tool_xtensa_elf_clang);
+        println!("Please, remove the directory before new installation.");
+    } else {
+        prepare_package_strip_prefix(&toolchain.llvm_url,
+                                     &toolchain.llvm_file,
+                                     toolchain.idf_tool_xtensa_elf_clang.clone(),
+                                     "xtensa-esp32-elf-clang"
+        );
+    }
+
+    println!("Updating environment variables:");
+    let libclang_bin = format!("{}/bin/", toolchain.idf_tool_xtensa_elf_clang);
+    println!("PATH+=\";{}\"", libclang_bin);
+    update_env_path(&libclang_bin);
```

Comment:
This PR does not really compile on non-windows platforms, because `update_env_path` is defined only on Windows...

Reply:
Thank you for notification. The non-Windows implementation will be updated.

Reply:
Project is now buildable on macOS. Added guess of triple to determine the package.
The tool supports extraction of .zip.
Support for .tar.xz should be added soon.

Reply:
I confirm that using `cargo install idf_env --git https://github.com/espressif/idf-env --branch feature/rust-install` is now building successfully on Mac.

As for not having support for tar/gzip - indeed. The app crashes shortly after issuing `idf-env rust reinstall` as it tries to "unzip" the `.tgz` - but you know that I guess. JFYI, sample [tar crate](https://crates.io/crates/tar).

Reply:
By the way you might want to rename the [package](https://github.com/espressif/idf-env/blob/main/Cargo.toml#L2) from `idf_env` to `idf-env` if you decide to publish the tool to `crates.io`, so that people are not confused that they have to do `cargo install idf_env`, but the installed binary tool is named `idf-env`.

I have some other general feedback, but not urgent at all.

Reply:
Typo in name fixed.